cirq-ft/cirq_ft/algos/arithmetic_gates.py

```diff
@@ -148,7 +149,14 @@ class BiQubitsMixer(infra.GateWithRegisters):
 
     @cached_property
     def signature(self) -> infra.Signature:
-        return infra.Signature.build(x=2, y=2, ancilla=3)
+        one_side = infra.Side.RIGHT if not self.adjoint else infra.Side.LEFT
+        return infra.Signature(
+            [
+                infra.Register('x', 2),
+                infra.Register('y', 2),
+                infra.Register('ancilla', 3, side=one_side),
```

Comment:
lets rename it `target`. this will result in some changes to the decomposition and docstring

Reply:
Let's also improve the docstring to more clearly specify what each of the 3 target bits contain. It's not clear to me right now by just reading the docstring.

Reply:
the short answer is .. enough information to do the adjoint. coming to think of it, this is why I kept the register name from the paper as ancilla. because they have no use otherthan as information holders for that will be used by the adjoint.

the actually useful information is stored inplace.

so maybe skip renaming it for now and I will come back for it later.

Reply:
hmm, we've renamed such registers to `junk` in Qualtran. 

@fdmalone As part of this PR, I think we should at least capture this discussion in the docstring and let the reader know that the ancilla is used to reduce the T-counts and assumes that it should be cleaned up later by the adjoint.

Reply:
Ok. I'll try distill something later today. We should open an issue to track either here or in qualtran.

Reply:
Done.